paddle/fluid/framework/ir/mkldnn/params_quantization_mkldnn_pass.cc

```diff
@@ -0,0 +1,219 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "paddle/fluid/framework/ir/mkldnn/params_quantization_mkldnn_pass.h"
+#include "paddle/fluid/framework/op_version_registry.h"
+#include "paddle/fluid/platform/mkldnn_helper.h"
+#include "paddle/fluid/string/pretty_log.h"
+
+namespace paddle {
+namespace framework {
+namespace ir {
+
+namespace {
+
+template <typename T_out>
+void QuantizeParams(LoDTensor* param_tensor, const std::vector<float>& scales) {
+  std::vector<T_out> tmp_data;
+  tmp_data.reserve(param_tensor->numel());
+
+  auto length = param_tensor->numel() / scales.size();
+
+  const float* param_data = param_tensor->data<float>();
+  for (int64_t i = 0; i < param_tensor->numel(); ++i) {
+    tmp_data[i] =
+        static_cast<T_out>(std::round(param_data[i] * scales[i / length]));
+  }
+
+  auto dims = param_tensor->dims();
+  param_tensor->clear();
+  param_tensor->Resize(dims);
+
+  auto int_param_data = param_tensor->mutable_data<T_out>(CPUPlace());
+  std::copy_n(tmp_data.data(), param_tensor->numel(), int_param_data);
+}
+
+ir::Node* FindOpInput(ir::Node* op, const std::string& input_name) {
+  auto op_input_it =
+      std::find_if(op->inputs.begin(), op->inputs.end(),
+                   [&](ir::Node* node) { return node->Name() == input_name; });
+
+  PADDLE_ENFORCE_NE(
+      op_input_it, op->inputs.end(),
+      platform::errors::InvalidArgument("Not found input %s of operator %s.",
+                                        input_name, op->Op()->Type()));
+  return (*op_input_it);
+}
+
+void ConnectInput(ir::Node* conv_op, const std::string& input_name,
+                  ir::Node* int_node) {
+  conv_op->Op()->SetInput(input_name, {int_node->Name()});
+  IR_NODE_LINK_TO(int_node, conv_op);
+}
+
+bool HasBias(ir::Node* conv_op) {
+  auto input_names = conv_op->Op()->InputNames();
+  return std::find(input_names.begin(), input_names.end(), "Bias") !=
+             input_names.end() &&
+         conv_op->Op()->Input("Bias").size() > 0;
+}
+
+bool ShouldSkipConv(ir::Node* conv_op, Scope* scope, ir::Node* conv_filter) {
+  if (!platform::HasOpINT8DataType(conv_op->Op())) {
+    return true;
+  }
+
+  auto filter_var = scope->GetVar(conv_filter->Name());
+  if (filter_var->Get<LoDTensor>().dtype() != phi::DataType::FLOAT32) {
+    VLOG(4) << "Skipping convolution (id: " << conv_op->id()
+            << ") because it's a bug that it is detected again.";
+    return true;
+  }
+
+  VLOG(4) << "Not skipping convolution (id: " << conv_op->id() << ")";
+  return false;
```

Comment:
You log when convolution is skipped and when it's not.
But you lack of consistency here, it will be confusing to debug.
Please add a VLOG to line 74.

Reply:
Ok